addons/field_rations/XEH_preInit.sqf

```diff
@@ -13,13 +13,25 @@ GVAR(thirstModifiers) = [];
 GVAR(hungerModifiers) = [];
 
 // List of p3d water sources (case sensitive) for objects that report `typeOf == ""`
-GVAR(waterSourceP3ds) = [
-    "misc_wellpump.p3d" // [11899, 9150, 0] on chenarus
-];
+GVAR(waterSourceP3Ds) = [];
 
 // List of refill action offsets corresponding to the p3ds in the array above
-GVAR(waterSourceOffsets) = [
-    [0, 0, 0]
-];
+GVAR(waterSourceOffsets) = [];
+
+// Fill water source arrays from CfgVehicles
+{
+    private _split = (getText (_x >> "model")) splitString "\";
+    private _string = toLower (_split select ((count _split) - 1));
+    _string = if ((_string find ".p3d") == -1) then {
```

Comment:
Could be:
```
if (_string find ".p3d" == -1) then {
    _string = _string + ".p3d";
};
```
But I think that is nitpicking and more code style

Reply:
Wouldn't it make more sense to just check the last 3/4 characters of the string? Instead of iterating over the whole string trying to find something that should only be at the end?

Reply:
Ah I couldn't come up with a nice way to get the end of the string so I just used `find`. Wouldn't it be slower with all those added commands though?

Reply:
No idea.. Would have to profile.